### PR TITLE
WebGLSender doesn't really implement Serializable

### PR DESCRIPTION
**Pull request by asajeffrey**
Add an API which turns a `WebGLSender` into something that is actually serializable.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because this is fixing a panic

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23746)

### CODE REVIEW

**Comment by CYBAI** [comment]
@bors-servo retry

- #21696

**Comment by Manishearth** [comment]
With this applied I still get the same crash

https://gist.github.com/Manishearth/8217b8d4f4684c044260472d6f3e057c

File: components/canvas_traits/webgl.rs [hunk 1]
```diff
@@ -175,10 +175,25 @@ impl WebGLMsgSender {
...
+#[derive(Clone, Debug, Deserialize, MallocSizeOf, Serialize)]
+struct SerializableWebGLMsgSender {
+    ctx_id: WebGLContextId,
+    #[ignore_malloc_size_of = "channels are hard"]
+    sender: IpcSender<WebGLMsg>,
 }
 
 #[typetag::serde]
-impl webxr_api::WebGLExternalImageApi for WebGLMsgSender {
+impl webxr_api::WebGLExternalImageApi for SerializableWebGLMsgSender {
```
Comment (Manishearth) [inline components/canvas_traits/webgl.rs]: You're still sending `sender`, a webgl_channel, over `self.sender`, an IPC channel here

Comment (Manishearth) [reply components/canvas_traits/webgl.rs]: We'll need a separate lock API that works exclusively with IPC

Comment (Manishearth) [reply components/canvas_traits/webgl.rs]: Wrote the fix and stuffed it in https://github.com/servo/servo/pull/23731

**Pull request merged**